src/components/DeploymentList.vue

```diff
@@ -0,0 +1,352 @@
+            <p-button small @click="clear">
+              Clear Filters
+            </p-button>
+          </template>
+        </PEmptyResults>
+        <PEmptyResults v-else>
+          <template #message>
+            <p-loading-icon />
+          </template>
+        </PEmptyResults>
+      </template>
+    </p-table>
+
+    <p-pager v-if="pages > 1" v-model:page="page" :pages="pages" />
+  </p-content>
+</template>
+
+<script lang="ts" setup>
+  import { TableColumn, media } from '@prefecthq/prefect-design'
+  import { NumberRouteParam, useDebouncedRef, useRouteQueryParam } from '@prefecthq/vue-compositions'
+  import { secondsInWeek } from 'date-fns/constants'
+  import { snakeCase } from 'lodash'
+  import merge from 'lodash.merge'
+  import { computed, ref } from 'vue'
+  import {
+    DeploymentsDeleteButton,
+    ResultsCount,
+    SearchInput,
+    FlowPopover,
+    MiniDeploymentHistory,
+    SelectedCount,
+    DeploymentTagsInput,
+    DeploymentToggle,
+    FormattedDate,
+    DeploymentStatusBadge
+  } from '@/components'
+  import { useCan, useDeploymentsFilterFromRoute, useWorkspaceRoutes, useDeployments, useComponent } from '@/compositions'
+  import { Deployment, DeploymentSchedule } from '@/models'
+  import { DeploymentsFilter } from '@/models/Filters'
+  import { ClassValue } from '@/types'
+  import { deploymentSortOptions } from '@/types/SortOptionTypes'
+
+  const props = defineProps<{
+    filter?: DeploymentsFilter,
+    prefix?: string,
+  }>()
+
+  const { DeploymentMenu } = useComponent()
+
+  const can = useCan()
+  const routes = useWorkspaceRoutes()
+
+  const nameLike = ref<string>()
+  const nameLikeDebounced = useDebouncedRef(nameLike, 1200)
+
+  const { filter, clear, isCustomFilter } = useDeploymentsFilterFromRoute(merge({}, props.filter, {
+    deployments: {
+      flowOrDeploymentNameLike: nameLikeDebounced,
+    },
+    limit: 50,
+  }), props.prefix)
+
+  const page = useRouteQueryParam('page', NumberRouteParam, 1)
+  const { deployments, subscriptions, total, pages } = useDeployments(filter, {
+    page,
+    interval: 30000,
+  })
+
+  const columns = computed<TableColumn<Deployment>[]>(() => [
+    {
+      label: 'Deployment',
+    },
+    {
+      label: 'Status',
+      property: 'status',
+      width: '116px',
+    },
+    {
+      label: 'Activity',
+      visible: media.lg,
+      maxWidth: '15%',
+    },
+    {
+      label: 'Tags',
+      property: 'tags',
+      visible: media.md,
+      maxWidth: '15%',
+    },
+    {
+      label: 'Schedules',
+      property: 'schedules',
+      visible: media.md,
+      maxWidth: '15%',
+    },
+    {
+      label: 'Action',
+      width: '82px',
+    },
+  ])
+
+  const columnClasses = (column: TableColumn<Deployment>): ClassValue => [`deployment-list__${snakeCase(column.label)}-column`]
+  const rowClasses = (row: Deployment): ClassValue => {
+    return [
+      {
+        'deployment-list__row--subdued': row.paused,
+      },
+    ]
```

Comment:
```suggestion
    return {
      'deployment-list__row--subdued': row.paused,
    }
```

Reply:
Ah I just realized the type for this is wrong as defined in ui-lib, it's requiring this to be an array of strings or records. Let me see if that's fixable.

Reply:
Ok I just copied what we had in prefect-design for now which makes a lot more sense

Reply:
ah right. Hmm I feel like this is more accurate if it still works
```
export type ClassValue = string | string[] | Record<string, boolean> | (string | Record<string, boolean>)[]
```

Reply:
ah yeah that one makes more sense too. Should probably delete the type from ui-library.

Reply:
Yeah that works for this component at least, let's see how the rest of the build behaves